src/System.IO.FileSystem/tests/Enumeration/MatchTypesTests.netcoreapp.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Tests.Enumeration
+{
+    public abstract class MatchTypesTests : FileSystemTest
+    {
+        protected abstract string[] GetPaths(string directory, string pattern, EnumerationOptions options);
+
+        [Fact]
+        public void QuestionMarkBehavior()
+        {
+            DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
+            FileInfo fileOne = new FileInfo(Path.Combine(testDirectory.FullName, "a.one"));
+            FileInfo fileTwo = new FileInfo(Path.Combine(testDirectory.FullName, "ab.two"));
+            FileInfo fileThree = new FileInfo(Path.Combine(testDirectory.FullName, "abc.three"));
+
+            fileOne.Create().Dispose();
+            fileTwo.Create().Dispose();
+            fileThree.Create().Dispose();
+
+            // Question marks collapse to periods in Win32
```

Comment:
Period in what sense? Not in the regex sense of if can match nothing

Reply:
Literal period

Reply:
I guess I don't know what collapse to period means.

Reply:
It means that question marks are optional if they but up contiguously against a period.  So a???.foo is actually a.foo | a?.foo | a??.foo | a???.foo. Evil that is attempted 8.3 matching matching. :P